ion/binarywriter.go

```diff
@@ -191,6 +192,11 @@ func (w *binaryWriter) WriteFloat(val float64) error {
 	bits := math.Float64bits(val)
 	binary.BigEndian.PutUint64(bs[1:], bits)
 
+	// The above PutUint64() puts an unwanted 1 as the last byte for NaN.
```

Comment:
Can you confirm ```NaN``` is always 8 bits? If ```NaN``` can be presented in 4 bits (float32) we might want to add the same logic on line 185 too.

Reply:
We know the last index is 8 because we have `bs := make([]byte, 9)` above.

Reply:
On [line 179](https://github.com/amzn/ion-go/pull/155/files#diff-2e0c44cac09fc56c07997526b83a087eR179) we check whether the value is a float32 or 64. If it is a float32 then we'll have ```make([]byte, 5)``` otherwise array byte of size 9.
My concern was if there is a case where ```NaN``` is float32

Reply:
Here the float32 case doesn't happen for NaN because we're getting a float64 value passed in and float64 NaN does not pass the `val == float64(float32(val))` condition.

Regardless I checked anyway and discovered that the unwanted 1 issue only applies to the 64 bit case.

Reply:
Thanks